server/api/cvrs.py

```diff
@@ -122,33 +123,81 @@ def are_uploaded_cvrs_valid(contest: Contest):
         return False
 
 
-# Wraps Jurisdiction.cvr_contest_metadata, applying any contest name
-# standardizations in Jurisdiction.contest_name_standardizations. This wrapper
-# should always be used for reading the metadata, so that the contest names
-# from the CVR will match those selected by the AA.
+# Wraps Jurisdiction.cvr_contest_metadata, applying any contest and choice name
+# standardizations. This wrapper should always be used for reading the
+# metadata, so that contest and choice names from CVR files will match those
+# provided by the AA.
 def cvr_contests_metadata(
-    jurisdiction: Jurisdiction,
+    jurisdiction: Jurisdiction, should_standardize_contest_choice_names=True
 ) -> Optional[CVR_CONTESTS_METADATA]:
     metadata = typing_cast(
         Optional[CVR_CONTESTS_METADATA], jurisdiction.cvr_contests_metadata
     )
     if metadata is None:
         return None
 
-    standardizations = typing_cast(
-        Optional[Dict[str, str]], jurisdiction.contest_name_standardizations
+    contest_name_standardizations = (
+        typing_cast(
+            Optional[Dict[str, Optional[str]]],
+            jurisdiction.contest_name_standardizations,
+        )
+        or {}
     )
-    standardizations = {
+    cvr_contest_name_to_standardized_contest_name = {
         cvr_contest_name: contest_name
-        for contest_name, cvr_contest_name in (standardizations or {}).items()
+        for contest_name, cvr_contest_name in contest_name_standardizations.items()
         if cvr_contest_name
     }
 
-    return {
-        standardizations.get(cvr_contest_name, cvr_contest_name): contest_metadata
-        for cvr_contest_name, contest_metadata in metadata.items()
+    contest_choice_name_standardizations = (
+        typing_cast(
+            Optional[Dict[str, Dict[str, Optional[str]]]],
+            jurisdiction.contest_choice_name_standardizations,
+        )
+        or {}
+    )
+    cvr_choice_name_to_standardized_choice_name_by_contest_id = {
+        contest_id: {
+            cvr_choice_name: choice_name
+            for cvr_choice_name, choice_name in standardizations.items()
+            if choice_name
+        }
+        for contest_id, standardizations in contest_choice_name_standardizations.items()
+    }
+
+    contest_name_to_id = {
+        contest.name: contest.id for contest in jurisdiction.election.contests
     }
 
+    standardized_metadata = {}
+    for cvr_contest_name, contest_metadata in metadata.items():
+        standardized_contest_name = cvr_contest_name_to_standardized_contest_name.get(
+            cvr_contest_name, cvr_contest_name
+        )
+
+        contest_id = contest_name_to_id.get(standardized_contest_name, None)
```

Comment:
I'm realizing that the variable name `standardized_contest_name` is a bit misleading since it's really a `potentially_standardized_contest_name`. As we iterate over CVR metadata, this could be a standardized contest name, a contest name that has yet to be standardized, or a contest name completely irrelevant to the audit altogether. In the latter two cases, `contest_id` is in fact `None`

Updated the variable name for clarity accordingly: https://github.com/votingworks/arlo/pull/1948/commits/8556ddd734b745bf32661dc445eb7884bbb8b876